ui/src/components/friends/incoming_requests/mod.rs

```diff
@@ -97,7 +97,8 @@ pub fn PendingFriends(cx: Scope) -> Element {
                 let _status_message = friend.status_message().unwrap_or_default();
                 let did = friend.did_key();
                 let did2 = did.clone();
-                let did_suffix: String = did.to_string().chars().rev().take(6).collect();
+                let mut did_suffix: String = did.to_string();
+                did_suffix = did_suffix.char_indices().nth_back(6).map(|(i,_)|did_suffix[i..].to_string()).unwrap_or_default();
```

Comment:
Instead, you can use `Identity::short_id` in which you could do something like

```
let did_suffix = friend.short_id().to_string();
```

Reply:
already in the description but it seems the short_id is not always present during requests?

Reply:
i concur. it can be done like this: 

```
let pk = did.public_key_bytes();
let short: [u8; 8] = pk[pk.len() - 8..].try_into().map_err(|_e| warp::error::Error::OtherWithContext("did to short".into()))?;
default.set_short_id(short);
```